pkg/scanners/scannerv4/nodescan_convert.go

```diff
@@ -113,3 +121,20 @@ func toStorageNotes(notes []v4.VulnerabilityReport_Note) []storage.NodeScan_Note
 	}
 	return convertedNotes
 }
+
+func fixNotes(notes []storage.NodeScan_Note, osImageRef string) []storage.NodeScan_Note {
+	isRHCOS := strings.HasPrefix(osImageRef, "Red Hat Enterprise Linux CoreOS")
+	if !isRHCOS {
+		// Don't change notes if scanned node isn't RHCOS
+		return notes
+	}
+	fixedNotes := make([]storage.NodeScan_Note, 0)
+	for _, note := range notes {
+		switch note {
+		case storage.NodeScan_UNSUPPORTED:
```

Comment:
So this one is being ignored? I find it strange, as we would land here only for RHCOS and we support node scan for it. So why do we need to filter this out? Have you seen it being set in some clusters?

Reply:
Yes, this is set for all RHCOS Nodes right now. 
The reason is that the VulnerabilityReport doesn't contain a `Distribution` entry.
Because said entry is missing, the Matcher will assign the `UNSUPPORTED` label [in here](https://github.com/stackrox/stackrox/blob/4deb7656a1dd421b173ddb6df6c9b5f50729f627/scanner/services/matcher.go#L148).

To correctly address this, we would need to fill the `Distributions` field in `IndexNode` ([here in Compliance](https://github.com/stackrox/stackrox/blob/5310f665f84d98a8a87f13f7fd7a3cf225a4dbab/compliance/index/v4/node.go#L69)).
To do that, we need to create an RHCOS DistributionScanner in ClairCore, which doesn't exist at this time.
I have created ROX-26593 to track this, as I'm very sure we cannot land this change in ClairCore before the code freeze.

Reply:
Makes sense! Could you capture this explanation in a comment if that isn't a problem?

Reply:
I left a TODO with the issue number on both, `fixNotes` and `toNodeScan`, which calls `fixNotes`.
In addition, I added a longer comment to `fixNotes` explaining the why and the fix.

Reply:
That looks great! Thank you